rfcs/2020-04-04-2221-kubernetes-integration.md

```diff
@@ -0,0 +1,1625 @@
+  - [OVH Managed Kubernetes Service](https://www.ovhcloud.com/en-gb/public-cloud/kubernetes/)
+  - [Rackspace Kubernetes-as-a-Service](https://www.rackspace.com/managed-kubernetes)
+  - [Linode Kubernetes Engine](https://www.linode.com/products/kubernetes/)
+  - [Yandex Managed Service for Kubernetes](https://cloud.yandex.com/services/managed-kubernetes)
+  - [Tencent Kubernetes Engine](https://intl.cloud.tencent.com/product/tke)
+- Kubernetes Distributions (for on-premise deployment)
+  - Production-grade
+    - bare `kubeadm`
+    - [OKD](https://www.okd.io/) (deploys OpenShift Origin)
+    - [Rancher Kubernetes Engine](https://rancher.com/products/rke/)
+    - [Metal3](http://metal3.io/)
+    - [Project Atomic Kubernetes](https://www.projectatomic.io/docs/kubernetes/)
+    - [Canonical Charmed Kubernetes](https://ubuntu.com/kubernetes/install#multi-node)
+    - [Kubernetes on DC/OS](https://github.com/mesosphere/dcos-kubernetes-quickstart)
+  - For small/dev deployments
+    - [Minikube](https://kubernetes.io/ru/docs/setup/learning-environment/minikube/)
+    - [MicroK8s](https://microk8s.io/)
+    - [Docker Desktop Kubernetes](https://www.docker.com/products/docker-desktop)
+    - [kind](https://kubernetes.io/docs/setup/learning-environment/kind/)
+    - [minishift](https://www.okd.io/minishift/)
+- [Container Runtimes (CRI impls)](https://kubernetes.io/docs/setup/production-environment/container-runtimes/)
+  - [Docker](https://www.docker.com/) (Kubernetes still has some "special"
+    integration with Docker; these days, "using Docker" technically means using
+    `runc` via `containerd` via `docker-engine`)
+  - OCI (via [CRI-O](https://cri-o.io/) or [containerd](https://containerd.io/))
+    - [runc](https://github.com/opencontainers/runc)
+    - [runhcs](https://github.com/Microsoft/hcsshim/tree/master/cmd/runhcs) -
+      see more [here][windows_in_kubernetes]
+    - [Kata Containers](https://github.com/kata-containers/runtime)
+    - [gVisor](https://github.com/google/gvisor)
+    - [Firecracker](https://github.com/firecracker-microvm/firecracker-containerd)
+
+We can't possibly expand this matrix densely due to the enormous amount of
+effort required to maintain the infrastructure and the costs. It may also be
+inefficient to test everything everywhere, because a lot of configurations
+don't have any significant or meaningful differences among each other.
+
+Testing various managed offers and distributions is not as important as testing
+different Kubernetes versions and container runtimes.
+
+It's probably a good idea to also test against the most famous managed
+Kubernetes provides: AWS, GCP and Azure. Just because our users are most likely
+to be on one of those.
+
+So, the goal for integration tests is to somehow test Vector with Kubernetes
+versions from MSKV to latest, all the container runtimes listed above and,
+additionally, on AWS, GCP and Azure.
```

Comment:
Agree. AWS, GCP, and Azure are good places to start, we can expand if we need after that.